nvtabular/column_group.py

```diff
@@ -179,27 +207,160 @@ def __getitem__(self, columns):
         """
         if isinstance(columns, str):
             columns = [columns]
+        if isinstance(columns, int):
+            return self.column_names[columns]
 
-        child = ColumnGroup(columns)
+        filtered_columns = [col for col in _convert_col(columns) if col.name in self.column_names]
+        child = ColumnGroup(filtered_columns)
         child.parents = [self]
         self.children.append(child)
         child.kind = str(columns)
         return child
 
+    def filter_columns(self, filter_fn, by_name=True):
+        if by_name:
+            filtered = [c for c in self.columns if filter_fn(c.name)]
+        else:
+            filtered = [c for c in self.columns if filter_fn(c)]
+
+        return self[filtered]
+
+    def filter_by_namespace(self, namespace):
+        return self.filter_columns(lambda c: c.startswith(namespace))
+
+    def get_tagged(self, tags, output_list=False, tags_to_filter=None):
```

Comment:
When would you want this method to return a `list` vs a `ColumnGroup`? What would be the impact of standardizing the output type to one or the other?

Reply:
See 248-249